pgxn/remotexact/remotexact.c

```diff
@@ -641,11 +641,21 @@ clean_up_xact_callback(XactEvent event, void *arg)
 	 */
 	MyProc->isRemoteXact = false;
 	pg_write_barrier();
-	if (event == XACT_EVENT_ABORT ||
-		event == XACT_EVENT_PARALLEL_ABORT ||
-		event == XACT_EVENT_COMMIT ||
-		event == XACT_EVENT_PARALLEL_COMMIT)
-		rwset_collection_buffer = NULL;
+
+	switch (event)
+	{
+		case XACT_EVENT_ABORT:
+		case XACT_EVENT_PARALLEL_ABORT:
+		case XACT_EVENT_COMMIT:
+		case XACT_EVENT_PARALLEL_COMMIT:
+		case XACT_EVENT_PREPARE:
```

Comment:
Previously, we didn't reset `rwset` after the transaction was prepared, so the next transaction would still use the old `rwset`